File: backend/test/java/de/griefed/serverpackcreator/utilities/commonutilities/FileUtilitiesTest.java
package de.griefed.serverpackcreator.utilities.commonutilities;

import org.apache.commons.io.FileUtils;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import java.io.File;
import java.io.IOException;

public class FileUtilitiesTest {

    private final FileUtilities FILE_UTILITIES;

    FileUtilitiesTest() {
        this.FILE_UTILITIES = new FileUtilities();
    }

    @Test
    void unzipArchiveTest() {
        String modpackDir = "backend/test/resources/curseforge_tests";
        String zipFile = "backend/test/resources/curseforge_tests/modpack.zip";
        FILE_UTILITIES.unzipArchive(zipFile, modpackDir);
        Assertions.assertTrue(new File("./backend/test/resources/curseforge_tests/manifest.json").exists());
        Assertions.assertTrue(new File("./backend/test/resources/curseforge_tests/modlist.html").exists());
        Assertions.assertTrue(new File("./backend/test/resources/curseforge_tests/overrides").isDirectory());
        FileUtils.deleteQuietly(new File("./backend/test/resources/curseforge_tests/manifest.json"));
        FileUtils.deleteQuietly(new File("./backend/test/resources/curseforge_tests/modlist.html"));
        FileUtils.deleteQuietly(new File("./backend/test/resources/curseforge_tests/overrides"));
    }

    @SuppressWarnings("ResultOfMethodCallIgnored")
    @Test
    void replaceFileTest() throws IOException {
        File source = new File("source.file");
        File destination = new File("destination.file");
        source.createNewFile();
        destination.createNewFile();
        FILE_UTILITIES.replaceFile(source,destination);
        Assertions.assertFalse(source.exists());
        Assertions.assertTrue(destination.exists());
        FileUtils.deleteQuietly(destination);
    }
}

File: serverpackcreator-api/src/jvmTest/kotlin/de/griefed/serverpackcreator/api/common/FileUtilitiesTest.kt
package de.griefed.serverpackcreator.api.common

import de.griefed.serverpackcreator.api.ApiWrapper
import de.griefed.serverpackcreator.api.utilities.common.FileUtilities
import de.griefed.serverpackcreator.api.utilities.common.deleteQuietly
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import java.io.File
import java.io.IOException

class FileUtilitiesTest internal constructor() {
    private var fileUtilities: FileUtilities =
        ApiWrapper.api(File("src/jvmTest/resources/serverpackcreator.properties")).fileUtilities

    @Test
    fun unzipArchiveTest() {
        val modpackDir = "src/jvmTest/resources/curseforge_tests"
        val zipFile = "src/jvmTest/resources/curseforge_tests/modpack.zip"
        fileUtilities.unzipArchive(zipFile, modpackDir)
        Assertions.assertTrue(
            File("src/jvmTest/resources/curseforge_tests/manifest.json").exists()
        )
        Assertions.assertTrue(
            File("src/jvmTest/resources/curseforge_tests/modlist.html").exists()
        )
        Assertions.assertTrue(
            File("src/jvmTest/resources/curseforge_tests/overrides").isDirectory
        )
        File("src/jvmTest/resources/curseforge_tests/manifest.json").deleteQuietly()
        File("src/jvmTest/resources/curseforge_tests/modlist.html").deleteQuietly()
        File("src/jvmTest/resources/curseforge_tests/overrides").deleteQuietly()
    }

    @Test
    @Throws(IOException::class)
    fun replaceFileTest() {
        val source = File("source.file")
        val destination = File("destination.file")
        source.createNewFile()
        destination.createNewFile()
        fileUtilities.replaceFile(source, destination)
        Assertions.assertFalse(source.exists())
        Assertions.assertTrue(destination.exists())
        destination.deleteQuietly()
    }
}